toolchain/sem_ir/typed_insts.h

```diff
@@ -256,8 +256,7 @@ struct AssociatedEntityType {
 
   TypeId type_id;
   // The interface in which the entity was declared.
-  // TODO: Consider storing a `SpecificInterfaceId` instead.
-  TypeId interface_type_id;
+  SpecificInterfaceId interface_id;
```

Comment:
Why a `SpecificInterfaceId` instead of an `InterfaceId` & `SpecificId` pair? Seems like we could avoid an indirection.

Reply:
My thinking was that as use of `SpecificInterfaceId` increased, we would find consumers wanted a `SpecificInterfaceId` instead of the pair of elements. But that doesn't seem to be the case right now; happy to switch it to the other form.

Reply:
Done. I added an accessor on the typed inst to get a `SpecificInterface` since coupling the two fields together seems generally to be an improvement.

Reply:
Yes, either is fine but I think Josh is right that we can avoid an indirection here until we need another argument here. The accessor seems like a good compromise, we can/should pass around SpecificInterface as a value.